File: java-sfn-unit-test/src/main/java/com/sfnut/myapp/parent/Step20_Nested_InputAdapter.java
package com.sfnut.myapp.parent;

import com.sfnut.sfn.nested.SfnTransform;
import com.sfnut.myapp.child.ChildPayload;

public class Step20_Nested_InputAdapter extends SfnTransform<ParentPayload, ChildPayload> {
    @Override
    public ChildPayload execute(ParentPayload input) {
        ChildPayload payload = new ChildPayload();
        payload.setHappyPathData(input.getHappyPathData());
        payload.setNestedFromParent(input.getNestedObject().getNestedField());

        return payload;
    }
}

File: myapp/parent/Step20_Nested_InputAdapter.cs
using sfn_ut.sfn.nested;
using sfn_ut.myapp.child;

namespace sfn_ut.myapp.parent;

public class Step20_Nested_InputAdapter : SfnTransform<ParentPayload, ChildPayload> {
  public override ChildPayload Execute(ParentPayload input) {
    return new ChildPayload(input.HappyPathData);
  }
}

